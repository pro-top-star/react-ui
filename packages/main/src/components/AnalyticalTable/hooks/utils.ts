export const makeTemplateColumns = (columns, resizedColumns) => {
  return columns
    .map((column) => {
      if (resizedColumns[column.id]) {
        return `${resizedColumns[column.id]}px`;
      }
      return `minmax(${column.minWidth}px, ${column.width || '1fr'})`;
    })
    .join(' ');
};
