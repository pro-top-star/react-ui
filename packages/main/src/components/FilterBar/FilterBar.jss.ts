import { JSSTheme } from '../../interfaces/JSSTheme';

const styles = ({ parameters }: JSSTheme) => {
  return {
    outerContainer: {
      paddingTop: '0.5rem',
      paddingLeft: '2rem',
      paddingRight: '2rem',
      paddingBottom: '1px',
      background: parameters.sapUiObjectHeaderBackground,
      boxShadow: parameters.sapUiShadowHeader
    },
    filterBarHeader: {
      alignItems: 'center',
      display: 'flex',
      background: parameters.sapUiObjectHeaderBackground,
      minHeight: '3rem',
      paddingBottom: '0.5rem',
      boxShadow: 'none',
      flexWrap: 'wrap'
    },
    vLine: {
      borderLeft: '1px solid gray',
      paddingLeft: '0.5rem'
    },
    filterArea: {
      display: 'flex',
      flexWrap: 'wrap',
      paddingTop: '1rem',
      paddingBottom: '1rem',
      background: parameters.sapUiObjectHeaderBackground,
      transition: 'max-height 0.2s ease-out, opacity 0.2s ease-in'
    },
    filterAreaClosed: {
      maxHeight: '0',
      opacity: 0,
      padding: 0,
      overflowY: 'auto'
    },
    filterAreaOpen: {
      maxHeight: '500px',
      opacity: 1,
      overflowY: 'auto'
    },
    headerRowRight: {
      display: 'flex',
      justifyContent: 'flex-end',
      flexGrow: 1
    },
    // is being applied to the span which represents the InfoLabel Text
    label: {
      fontSize: parameters.sapMFontSmallSize,
      fontFamily: parameters.sapUiFontFamily,
      lineHeight: '1.125rem',
      fontWeight: 600,
      letterSpacing: '0.0125rem',
      textTransform: 'uppercase',
      textAlign: 'center',
      verticalAlign: 'top',
      textOverflow: 'ellipsis',
      whiteSpace: 'nowrap',
      display: 'inline-block',
      color: parameters.sapUiBaseText
    },
    // specific padding needed for purely numeric input
    numeric: {},
    // specific padding needed for text input
    text: {},
    // displayOnly mode
    displayOnly: {}
  };
};

export default styles;
