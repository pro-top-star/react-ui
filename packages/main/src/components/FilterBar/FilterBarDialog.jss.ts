import { CssSizeVariables } from '@ui5/webcomponents-react-base/lib/CssSizeVariables';
import { sapUiContentPadding } from '@ui5/webcomponents-react-base/lib/spacing';

const styles = {
  dialog: {
    ...sapUiContentPadding,
    display: 'flex',
    flexDirection: 'column',
    maxWidth: '960px',
    width: '80vw',
    maxHeight: '70vh'
  },
  header: {
    padding: '0.25rem 1rem 0 1rem',
    '& *': {
      margin: '0.25rem 0 0.25rem 0'
    },
    '& ui5-input': {
      width: '100%'
    }
  },
  footer: {
    '& :not(:last-child)': {
      marginRight: '0.25rem'
    }
  },
  groupContainer: {
    display: 'flex',
    flexDirection: 'column'
  },
  groupTitle: {
    maxWidth: '85%',
    marginRight: '0.5rem'
  },
  filters: {
    padding: '1rem 0 2rem 0'
  },
  singleFilter: {
    display: 'grid',
    gridTemplateColumns: `auto minmax(${CssSizeVariables.sapWcrCheckBoxWidthHeight},7%)`,
    gridTemplateRows: 'auto',
    gridColumnGap: '0.5rem',
    '@media(max-width:700px)': {
      marginTop: '0.5rem'
    },
    '& ui5-checkbox': {
      placeSelf: 'center start',
      '@media(max-width:700px)': {
        marginTop: '0.8rem',
        paddingLeft: 0,
        placeSelf: 'end start'
      }
    }
  },

  fbSearch: {
    '@media(min-width:700px)': {
      display: 'grid',
      gridTemplateColumns: '20% auto 7%',
      gridTemplateRows: 'auto',
      gridRowGap: '0.5rem',
      gridColumnGap: '0.5rem'
    },
    paddingBottom: '2rem',
    width: '100%',
    '& ui5-input': {
      width: '100%'
    }
  }
};

export default styles;
