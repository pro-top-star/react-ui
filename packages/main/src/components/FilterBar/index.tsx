import { StyleClassHelper } from '@ui5/webcomponents-react-base/lib/StyleClassHelper';
import { usePassThroughHtmlProps } from '@ui5/webcomponents-react-base/lib/usePassThroughHtmlProps';
import { Button } from '@ui5/webcomponents-react/lib/Button';
import { ButtonDesign } from '@ui5/webcomponents-react/lib/ButtonDesign';
import React, { FC, forwardRef, ReactNode, ReactNodeArray, RefObject, useCallback, useState } from 'react';
import { createUseStyles } from 'react-jss';
import { ClassProps } from '../../interfaces/ClassProps';
import { CommonProps } from '../../interfaces/CommonProps';
import styles from './FilterBar.jss';

export interface FilterBarPropTypes extends CommonProps {
  renderVariants?: () => JSX.Element;
  renderSearch?: () => JSX.Element;
  children: ReactNode | ReactNodeArray;
}

interface FilterBarInternalProps extends FilterBarPropTypes, ClassProps {}

const useStyles = createUseStyles<keyof ReturnType<typeof styles>>(styles, { name: 'FilterBar' });

/**
 * <code>import { FilterBar } from '@ui5/webcomponents-react/lib/FilterBar';</code>
 */
const FilterBar: FC<FilterBarPropTypes> = forwardRef((props: FilterBarPropTypes, ref: RefObject<HTMLDivElement>) => {
  const { children, renderVariants, renderSearch } = props as FilterBarInternalProps;
  const [showFilters, setShowFilters] = useState(true);

  const classes = useStyles();

  const handleHideFilterBar = useCallback(() => {
    setShowFilters(!showFilters);
  }, [showFilters]);

  const filterAreaClasses = StyleClassHelper.of(classes.filterArea);
  if (showFilters) {
    filterAreaClasses.put(classes.filterAreaOpen);
  } else {
    filterAreaClasses.put(classes.filterAreaClosed);
  }

  const passThroughProps = usePassThroughHtmlProps(props);

  return (
    <div ref={ref} className={classes.outerContainer} {...passThroughProps}>
      <div className={classes.filterBarHeader}>
        {renderVariants && renderVariants()}
        {renderSearch && <div className={classes.vLine}> {renderSearch()} </div>}
        <div className={classes.headerRowRight}>
          <Button onClick={handleHideFilterBar} design={ButtonDesign.Transparent}>
            {showFilters ? 'Hide Filter Bar' : 'Show Filter Bar'}
          </Button>
        </div>
      </div>
      <div className={filterAreaClasses.valueOf()}>{children}</div>
    </div>
  );
});

FilterBar.defaultProps = {
  children: ''
};

FilterBar.displayName = 'FilterBar';
export { FilterBar };
