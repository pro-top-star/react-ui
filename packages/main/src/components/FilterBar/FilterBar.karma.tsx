import { mountThemedComponent } from '@shared/tests/utils';
import { expect, use } from 'chai';
import { matchSnapshot } from 'chai-karma-snapshot';
import React from 'react';
import { FilterBar } from '../../lib/FilterBar';
import { FilterItem } from '../../lib/FilterItem';
import { FilterType } from '../../lib/FilterType';
import { Input } from '../../lib/Input';
import { Switch } from '../../lib/Switch';
import { VariantManagement } from '../../lib/VariantManagement';

const variantItems = [{ label: 'Variant 1', key: '1' }, { label: 'Variant 2', key: '2' }];
const filterItems = [{ text: 'Text 1', key: '1' }, { text: 'Text 2', key: '2' }];

const renderVariants = () => <VariantManagement variantItems={variantItems} />;
const renderSearch = () => <Input placeholder={'Search'} />;

use(matchSnapshot);

describe('FilterBar', () => {
  it('Render without crashing', () => {
    const wrapper = mountThemedComponent(
      <FilterBar renderSearch={renderSearch} renderVariants={renderVariants}>
        <FilterItem
          // onChange={(e) => alert(e.getParameter('selectedItem').key)}
          filterItems={filterItems}
          label="Classification"
          key="classification"
          type={FilterType.Select}
        />
        <FilterItem
          key={'filter2'}
          type={FilterType.Custom}
          label={'Custom Filter 1'}
          changeEventName={'onClick'}
          valueParamName={'state'}
        >
          <Switch />
        </FilterItem>
      </FilterBar>
    );
    expect(wrapper.debug()).to.matchSnapshot();
  });

  it('Hide Filter Bar', () => {
    const wrapper = mountThemedComponent(
      <FilterBar renderVariants={renderVariants}>
        <FilterItem
          // onChange={(e) => alert(e.getParameter('selectedItem').key)}
          filterItems={filterItems}
          label="Classification"
          key="classification"
          type={FilterType.Select}
        />
        <FilterItem
          key={'filter2'}
          type={FilterType.Custom}
          label={'Custom Filter 1'}
          changeEventName={'onClick'}
          valueParamName={'state'}
        >
          <Switch />
        </FilterItem>
        <FilterItem
          loading
          key={'filter3'}
          type={FilterType.Custom}
          label={'Custom Filter 1'}
          changeEventName={'onClick'}
          valueParamName={'state'}
        >
          <Switch />
        </FilterItem>
      </FilterBar>
    );
    const component = wrapper
      .find('ui5-button')
      .first()
      .instance() as any;
    component.fireEvent('click');
    expect(wrapper.debug()).to.matchSnapshot();
  });

  it('Select Filter Item', () => {
    const wrapper = mountThemedComponent(
      <FilterBar renderVariants={renderVariants}>
        <FilterItem
          // onChange={(e) => alert(e.getParameter('selectedItem').key)}
          filterItems={filterItems}
          label="Classification"
          key="classification"
          type={FilterType.Select}
        />
        <FilterItem
          // onChange={(e) => alert(e.getParameter('selectedItem').key)}
          loading
          filterItems={filterItems}
          label="Classification"
          key="classification2"
          type={FilterType.Select}
        />
        <FilterItem
          // onChange={(e) => alert(e.getParameter('selectedItem').key)}
          // filterItems={filterItems}
          label="Classification"
          key="classification3"
          type={FilterType.Default}
        />
        <FilterItem
          // onChange={(e) => alert(e.getParameter('selectedItem').key)}
          filterItems={filterItems}
          label="Classification"
          key="classification3"
          type={FilterType.MultiSelect}
        />
      </FilterBar>
    );
    // console.log(wrapper.debug());
    // console.log(wrapper.find('Select').debug());
    wrapper
      .find('ui5-option')
      .at(1)
      .simulate('change');

    expect(wrapper.debug()).to.matchSnapshot();
  });
});
