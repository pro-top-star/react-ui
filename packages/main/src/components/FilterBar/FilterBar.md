```jsx
import { FilterBar } from '@ui5/webcomponents-react/lib/FilterBar';
```
