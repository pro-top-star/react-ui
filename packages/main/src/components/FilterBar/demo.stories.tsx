import { action } from '@storybook/addon-actions';
import { boolean, select } from '@storybook/addon-knobs';
import { FilterBar } from '@ui5/webcomponents-react/lib/FilterBar';
import { FilterItem } from '@ui5/webcomponents-react/lib/FilterItem';
import { FilterType } from '@ui5/webcomponents-react/lib/FilterType';
import { Input } from '@ui5/webcomponents-react/lib/Input';
import { PlacementType } from '@ui5/webcomponents-react/lib/PlacementType';
import { Switch } from '@ui5/webcomponents-react/lib/Switch';
import { TitleLevel } from '@ui5/webcomponents-react/lib/TitleLevel';
import { VariantManagement } from '@ui5/webcomponents-react/lib/VariantManagement';
import React from 'react';
import notes from './FilterBar.md';

const variantItems = [
  { label: 'Variant 1', key: '1' },
  { label: 'Variant 2', key: '2' }
];
const filterItems = [
  { text: 'Text 1', key: '1' },
  { text: 'Text 2', key: '2' }
];

const renderVariants = () => {
  return (
    <VariantManagement
      style={{ width: '300px', height: 'auto' }}
      closeOnItemSelect={boolean('closeOnItemSelect', true)}
      initialSelectedKey={'2'}
      variantItems={variantItems}
      onSelect={(e) => {
        console.log(e.getParameter('selectedItem').key);
      }}
      placement={select('Placement', PlacementType, PlacementType.Bottom)}
      level={select('level', TitleLevel, TitleLevel.H6)}
    />
  );
};

const renderSearch = () => {
  return <Input placeholder={'Search'} />;
};

export const renderStory = () => {
  return (
    <FilterBar renderSearch={renderSearch} renderVariants={renderVariants}>
      <FilterItem
        onChange={(e) => alert(e.getParameter('selectedItem').key)}
        filterItems={filterItems}
        label="Classification"
        key="classification"
        type={FilterType.Select}
        loading={boolean('loading', false)}
      />
      <FilterItem
        key={'filter2'}
        type={FilterType.Custom}
        label={'Custom Filter 1'}
        changeEventName={'onClick'}
        valueParamName={'state'}
      >
        <Switch />
      </FilterItem>
      <FilterItem
        onChange={action('onChange')}
        filterItems={filterItems}
        style={{
          maxWidth: '200px'
        }}
        label="Multi"
        key="Multi"
        type={FilterType.MultiSelect}
        loading={boolean('loading', false)}
      />
    </FilterBar>
  );
};
renderStory.story = {
  name: 'Default'
};

export default {
  title: 'Components / FilterBar',
  component: FilterBar,
  parameters: { notes }
};
