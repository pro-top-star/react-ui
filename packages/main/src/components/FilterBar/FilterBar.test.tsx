import { createPassThroughPropsTest, mountThemedComponent } from '@shared/tests/utils';
import { FilterBar } from '@ui5/webcomponents-react/lib/FilterBar';
import { FilterItem } from '@ui5/webcomponents-react/lib/FilterItem';
import { FilterType } from '@ui5/webcomponents-react/lib/FilterType';
import { Input } from '@ui5/webcomponents-react/lib/Input';
import { Switch } from '@ui5/webcomponents-react/lib/Switch';
import { VariantManagement } from '@ui5/webcomponents-react/lib/VariantManagement';
import React from 'react';

const variantItems = [
  { label: 'Variant 1', key: '1' },
  { label: 'Variant 2', key: '2' }
];
const filterItems = [
  { text: 'Text 1', key: '1' },
  { text: 'Text 2', key: '2' }
];

const renderVariants = () => <VariantManagement variantItems={variantItems} />;
const renderSearch = () => <Input placeholder={'Search'} />;

describe('FilterBar', () => {
  it('Render without crashing', () => {
    const wrapper = mountThemedComponent(
      <FilterBar renderSearch={renderSearch} renderVariants={renderVariants}>
        <FilterItem
          // onChange={(e) => alert(e.getParameter('selectedItem').key)}
          filterItems={filterItems}
          label="Classification"
          key="classification"
          type={FilterType.Select}
        />
        <FilterItem
          key={'filter2'}
          type={FilterType.Custom}
          label={'Custom Filter 1'}
          changeEventName={'onClick'}
          valueParamName={'state'}
        >
          <Switch />
        </FilterItem>
      </FilterBar>
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('Hide Filter Bar', () => {
    const wrapper = mountThemedComponent(
      <FilterBar renderVariants={renderVariants}>
        <FilterItem
          // onChange={(e) => alert(e.getParameter('selectedItem').key)}
          filterItems={filterItems}
          label="Classification"
          key="classification"
          type={FilterType.Select}
        />
        <FilterItem
          key={'filter2'}
          type={FilterType.Custom}
          label={'Custom Filter 1'}
          changeEventName={'onClick'}
          valueParamName={'state'}
        >
          <Switch />
        </FilterItem>
        <FilterItem
          loading
          key={'filter3'}
          type={FilterType.Custom}
          label={'Custom Filter 1'}
          changeEventName={'onClick'}
          valueParamName={'state'}
        >
          <Switch />
        </FilterItem>
      </FilterBar>
    );
    const component = wrapper
      .find('ui5-button')
      .first()
      .instance() as any;
    component.fireEvent('click');
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('Select Filter Item', () => {
    const wrapper = mountThemedComponent(
      <FilterBar renderVariants={renderVariants}>
        <FilterItem
          // onChange={(e) => alert(e.getParameter('selectedItem').key)}
          filterItems={filterItems}
          label="Classification"
          key="classification"
          type={FilterType.Select}
        />
        <FilterItem
          // onChange={(e) => alert(e.getParameter('selectedItem').key)}
          loading
          filterItems={filterItems}
          label="Classification"
          key="classification2"
          type={FilterType.Select}
        />
        <FilterItem
          // onChange={(e) => alert(e.getParameter('selectedItem').key)}
          // filterItems={filterItems}
          label="Classification"
          key="classification3"
          type={FilterType.Default}
        />
        <FilterItem
          // onChange={(e) => alert(e.getParameter('selectedItem').key)}
          filterItems={filterItems}
          label="Classification"
          key="classification3"
          type={FilterType.MultiSelect}
        />
      </FilterBar>
    );
    // console.log(wrapper.render());
    // console.log(wrapper.find('Select').render());
    wrapper
      .find('ui5-option')
      .at(1)
      .simulate('change');

    expect(wrapper.render()).toMatchSnapshot();
  });

  createPassThroughPropsTest(FilterBar);
});
