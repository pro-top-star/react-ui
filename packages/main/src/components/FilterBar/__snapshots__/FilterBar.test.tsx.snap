// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBar Hide Filter Bar 1`] = `
<div
  class="FilterBar--outerContainer-"
>
  <div
    class="FilterBar--filterBarHeader-"
  >
    <div
      style="display: inline-block; pointer-events: auto;"
    >
      <div
        class="VariantManagement--VariantManagement-"
      >
        <ui5-title
          class="VariantManagement--VariantManagementText-"
          level="H4"
        >
          Variant 1
        </ui5-title>
        <ui5-button
          design="Transparent"
          icon="navigation-down-arrow"
        />
      </div>
    </div>
    <ui5-popover
      header-text="Variants"
      horizontal-align="Center"
      placement-type="Bottom"
      vertical-align="Center"
    >
      <ui5-button
        class="VariantManagement--footer-"
        design="Emphasized"
        slot="footer"
      >
        Cancel
      </ui5-button>
      <ui5-list
        footer-text=""
        header-text=""
        mode="SingleSelect"
        no-data-text=""
        separators="All"
      >
        <ui5-li
          data-key="1"
          info-state="None"
          selected="true"
          style="width: 300px;"
          type="Active"
        >
          Variant 1
        </ui5-li>
        <ui5-li
          data-key="2"
          info-state="None"
          style="width: 300px;"
          type="Active"
        >
          Variant 2
        </ui5-li>
      </ui5-list>
    </ui5-popover>
    <div
      class="FilterBar--headerRowRight-"
    >
      <ui5-button
        design="Transparent"
      >
        Hide Filter Bar
      </ui5-button>
    </div>
  </div>
  <div
    class="FilterBar--filterArea- FilterBar--filterAreaOpen-"
  >
    <div
      class="FilterItem--filterItem-"
    >
      <div
        class="FilterItem--innerFilterItemContainer-"
      >
        <ui5-label
          for=""
        >
          Classification
        </ui5-label>
        <ui5-select
          style="width: 100%;"
          value-state="None"
        >
          <ui5-option
            data-key="1"
          >
            Text 1
          </ui5-option>
          <ui5-option
            data-key="2"
          >
            Text 2
          </ui5-option>
        </ui5-select>
      </div>
    </div>
    <div
      class="FilterItem--filterItem-"
    >
      <div
        class="FilterItem--innerFilterItemContainer-"
      >
        <ui5-label
          for=""
        >
          Custom Filter 1
        </ui5-label>
        <div>
          <ui5-switch
            style="width: 100%;"
            value-parameter="state"
          />
        </div>
      </div>
    </div>
    <div
      class="FilterItem--filterItem-"
    >
      <div
        class="FilterItem--innerFilterItemContainer-"
      >
        <ui5-label
          for=""
        >
          Custom Filter 1
        </ui5-label>
        <div
          class="FilterItem--loadingContainer-"
        >
          <ui5-busyindicator
            active="true"
            size="Medium"
            style="background-color: transparent; width: 80px;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`FilterBar Render without crashing 1`] = `
<div
  class="FilterBar--outerContainer-"
>
  <div
    class="FilterBar--filterBarHeader-"
  >
    <div
      style="display: inline-block; pointer-events: auto;"
    >
      <div
        class="VariantManagement--VariantManagement-"
      >
        <ui5-title
          class="VariantManagement--VariantManagementText-"
          level="H4"
        >
          Variant 1
        </ui5-title>
        <ui5-button
          design="Transparent"
          icon="navigation-down-arrow"
        />
      </div>
    </div>
    <ui5-popover
      header-text="Variants"
      horizontal-align="Center"
      placement-type="Bottom"
      vertical-align="Center"
    >
      <ui5-button
        class="VariantManagement--footer-"
        design="Emphasized"
        slot="footer"
      >
        Cancel
      </ui5-button>
      <ui5-list
        footer-text=""
        header-text=""
        mode="SingleSelect"
        no-data-text=""
        separators="All"
      >
        <ui5-li
          data-key="1"
          info-state="None"
          selected="true"
          style="width: 300px;"
          type="Active"
        >
          Variant 1
        </ui5-li>
        <ui5-li
          data-key="2"
          info-state="None"
          style="width: 300px;"
          type="Active"
        >
          Variant 2
        </ui5-li>
      </ui5-list>
    </ui5-popover>
    <div
      class="FilterBar--vLine-"
    >
       
      <ui5-input
        placeholder="Search"
        type="Text"
        value=""
        value-state="None"
      />
       
    </div>
    <div
      class="FilterBar--headerRowRight-"
    >
      <ui5-button
        design="Transparent"
      >
        Hide Filter Bar
      </ui5-button>
    </div>
  </div>
  <div
    class="FilterBar--filterArea- FilterBar--filterAreaOpen-"
  >
    <div
      class="FilterItem--filterItem-"
    >
      <div
        class="FilterItem--innerFilterItemContainer-"
      >
        <ui5-label
          for=""
        >
          Classification
        </ui5-label>
        <ui5-select
          style="width: 100%;"
          value-state="None"
        >
          <ui5-option
            data-key="1"
          >
            Text 1
          </ui5-option>
          <ui5-option
            data-key="2"
          >
            Text 2
          </ui5-option>
        </ui5-select>
      </div>
    </div>
    <div
      class="FilterItem--filterItem-"
    >
      <div
        class="FilterItem--innerFilterItemContainer-"
      >
        <ui5-label
          for=""
        >
          Custom Filter 1
        </ui5-label>
        <div>
          <ui5-switch
            style="width: 100%;"
            value-parameter="state"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`FilterBar Select Filter Item 1`] = `
<div
  class="FilterBar--outerContainer-"
>
  <div
    class="FilterBar--filterBarHeader-"
  >
    <div
      style="display: inline-block; pointer-events: auto;"
    >
      <div
        class="VariantManagement--VariantManagement-"
      >
        <ui5-title
          class="VariantManagement--VariantManagementText-"
          level="H4"
        >
          Variant 1
        </ui5-title>
        <ui5-button
          design="Transparent"
          icon="navigation-down-arrow"
        />
      </div>
    </div>
    <ui5-popover
      header-text="Variants"
      horizontal-align="Center"
      placement-type="Bottom"
      vertical-align="Center"
    >
      <ui5-button
        class="VariantManagement--footer-"
        design="Emphasized"
        slot="footer"
      >
        Cancel
      </ui5-button>
      <ui5-list
        footer-text=""
        header-text=""
        mode="SingleSelect"
        no-data-text=""
        separators="All"
      >
        <ui5-li
          data-key="1"
          info-state="None"
          selected="true"
          style="width: 300px;"
          type="Active"
        >
          Variant 1
        </ui5-li>
        <ui5-li
          data-key="2"
          info-state="None"
          style="width: 300px;"
          type="Active"
        >
          Variant 2
        </ui5-li>
      </ui5-list>
    </ui5-popover>
    <div
      class="FilterBar--headerRowRight-"
    >
      <ui5-button
        design="Transparent"
      >
        Hide Filter Bar
      </ui5-button>
    </div>
  </div>
  <div
    class="FilterBar--filterArea- FilterBar--filterAreaOpen-"
  >
    <div
      class="FilterItem--filterItem-"
    >
      <div
        class="FilterItem--innerFilterItemContainer-"
      >
        <ui5-label
          for=""
        >
          Classification
        </ui5-label>
        <ui5-select
          style="width: 100%;"
          value-state="None"
        >
          <ui5-option
            data-key="1"
          >
            Text 1
          </ui5-option>
          <ui5-option
            data-key="2"
          >
            Text 2
          </ui5-option>
        </ui5-select>
      </div>
    </div>
    <div
      class="FilterItem--filterItem-"
    >
      <div
        class="FilterItem--innerFilterItemContainer-"
      >
        <ui5-label
          for=""
        >
          Classification
        </ui5-label>
        <div
          class="FilterItem--loadingContainer-"
        >
          <ui5-busyindicator
            active="true"
            size="Medium"
            style="background-color: transparent; width: 80px;"
          />
        </div>
      </div>
    </div>
    <div
      class="FilterItem--filterItem-"
    >
      <div
        class="FilterItem--innerFilterItemContainer-"
      >
        <ui5-label
          for=""
        >
          Classification
        </ui5-label>
        <ui5-input
          placeholder=""
          style="width: 100%;"
          type="Text"
          value=""
          value-state="None"
        />
      </div>
    </div>
    <div
      class="FilterItem--filterItem-"
    >
      <div
        class="FilterItem--innerFilterItemContainer-"
      >
        <ui5-label
          for=""
        >
          Classification
        </ui5-label>
        <ui5-multi-combobox
          placeholder=""
          value=""
          value-state="None"
        >
          <ui5-li
            data-key="1"
            info-state="None"
            type="Active"
          >
            Text 1
          </ui5-li>
          <ui5-li
            data-key="2"
            info-state="None"
            type="Active"
          >
            Text 2
          </ui5-li>
        </ui5-multi-combobox>
      </div>
    </div>
  </div>
</div>
`;
